src/groups.jl

```diff
@@ -1,31 +1,33 @@
-## Iterator protocol for `Group`
+################################################################################
+#
+#   groups.jl : Interface for group parents
+#
+################################################################################
 
-Base.eltype(::Type{G}) where {G<:Group} =
-    throw(InterfaceNotImplemented(:Iteration, "Base.eltype(::Type{$G})"))
-Base.iterate(G::Group) =
-    throw(InterfaceNotImplemented(:Iteration, "Base.iterate(::$(typeof(G)))"))
-Base.iterate(G::Group, state) = throw(
-    InterfaceNotImplemented(:Iteration, "Base.iterate(::$(typeof(G)), state)"),
-)
-Base.IteratorSize(::Type{<:Group}) = Base.SizeUnknown()
-Base.length(G::Group) = order(Int, G)
+# Creator : Marek Kaluba
 
-## `Group` interface
+################################################################################
+# Parents and elements
+################################################################################
 
-### Obligatory methods for `Group`
 @doc Markdown.doc"""
-    one(G::Group)
-Return the identity of the group.
+    eltype(::Type{G}) where {G <: Group}
+
+Alias for `elem_type`.
 """
+Base.eltype(::Type{G}) where {G <: Group} = elem_type(G)
```

Comment:
I would also restrain from adding a docstring to it; it's a `Base` function and has exactly one docstring, I think it's enough.

Reply:
I'm curious what is wrong with `elem_type`? I am all for defining our own method for getting element types and using `Base` instead of `AbstractAlgebra`, but I do not like the name `eltype`; `elem_type` better describes the method in my point of view.